### DISCUSSION TITLE
Method to maintain a session with httpx similar to requests

### DISCUSSION THREAD

**Discussion by mossan0101** [encode/httpx#2887] [Q&A] [answered]
Hello, 
I'm considering using httpx to download videos from a certain streaming service. 
Currently, I'm using requests and it's working well with 
### session = requests.Session() 
for login authentication and such. 
From what I've researched, it seems that you can manipulate sessions with 
### client = httpx.Client()
 in httpx as well. 
Looking at the Advanced Usage, it's recommended to close the session using context managers or close(), but since I’m going to be downloading videos repeatedly, I need to use a cookie that contains a token that changes every time, I want to maintain the session for a long time. Currently, with requests, it's working fine without closing the session until the program ends. 

Sorry for the long introduction, 
but here's my question:   Is it okay not to close the session until the program ends in order to keep the session in httpx? 

I’m using AI for translation. I apologize if it’s hard to read!

**Comment by karpetrosyan** [comment, marked as answer]
Hi! 
If you want to use the client constantly, it is completely okay to create it and never close it.

**Reply by mossan0101** [reply]
I’m relieved to hear that! Thank you for the answer!
